code/modules/food_and_drinks/pizzabox.dm

```diff
@@ -117,35 +119,38 @@
 		. += M
 
 /obj/item/pizzabox/attack_self(mob/user)
-	if(boxes.len > 0)
+	if(length(boxes) > 0)
 		return
 	open = !open
 	if(open && !bomb_defused)
 		audible_message(span_warning("[icon2html(src, hearers(src))] *beep*"))
 		bomb_active = TRUE
 		START_PROCESSING(SSobj, src)
-	else if(!open && !pizza && !bomb)
-		var/obj/item/stack/sheet/cardboard/cardboard = new /obj/item/stack/sheet/cardboard(user.drop_location())
-		to_chat(user, span_notice("You fold [src] into [cardboard]."))
-		user.put_in_active_hand(cardboard)
-		qdel(src)
-		return
 	update_appearance()
 
+/obj/item/pizzabox/attack_self_secondary(mob/user)
+	if(length(boxes) > 0)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(pizza || bomb || !foldable)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	var/obj/item/stack/sheet/cardboard/cardboard = new(user.drop_location())
+	user.put_in_active_hand(cardboard)
+	qdel(src)
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
+
 //ATTACK HAND IGNORING PARENT RETURN VALUE
 /obj/item/pizzabox/attack_hand(mob/user, list/modifiers)
 	if(user.get_inactive_held_item() != src)
 		return ..()
 	if(open)
 		if(pizza)
 			user.put_in_hands(pizza)
-			to_chat(user, span_notice("You take [pizza] out of [src]."))
 			pizza = null
 			update_appearance()
 		else if(bomb)
 			if(wires.is_all_cut() && bomb_defused)
 				user.put_in_hands(bomb)
-				to_chat(user, span_notice("You carefully remove the [bomb] from [src]."))
+				balloon_alert(user, "removed bomb")
```

Comment:
this would balloon on the box, so if it's in their hand, it would balloon in the icon in their UI.
Though it might not be a problem since that's likely where the user would be looking, but if that's not what you want, you'll have to change it to ``user.balloon_alert``

Reply:
ive just tested it again, the balloon alert just comes out of the guy

![image](https://user-images.githubusercontent.com/116288367/202826773-c421201d-f010-40c4-8a38-35cbf38081f6.png)

Reply:
It uses the turf of whatever it's on, it only shows in hand if it's not equipped...I think? I forget when exactly it does but bla